airbyte-integrations/connectors/source-gitlab/source_gitlab/source.py

```diff
@@ -186,14 +192,14 @@ def check_connection(self, logger, config) -> Tuple[bool, Any]:
             return True, None  # in case there's no projects
         except HTTPError as http_error:
             if config["credentials"]["auth_type"] == "oauth2.0":
-                if http_error.response.status_code == 401:
+                if http_error.response.status_code in (400, 401):
```

Comment:
If we hit the `400` it means the `refresh_token` is no longer valid, we should raise an Exception here with the action step to `Proceed with the `re-authentication` ....`. 
The `401` comes up when the `refresh_token` is valid, but the `access_token` was revoked or expired and should be refreshed.